stubs/urllib3/urllib3/util/retry.pyi

```diff
@@ -11,17 +12,33 @@ ResponseError = exceptions.ResponseError
 
 log: Any
 
+class RequestHistory(NamedTuple):
+    method: str | None
+    url: str | None
+    error: Exception | None
+    status: int | None
+    redirect_location: str | None
+
 class Retry:
-    DEFAULT_METHOD_WHITELIST: Any
-    BACKOFF_MAX: Any
-    total: Any
-    connect: Any
-    read: Any
-    redirect: Any
-    status_forcelist: Any
-    method_whitelist: Any
-    backoff_factor: Any
-    raise_on_redirect: Any
+    DEFAULT_ALLOWED_METHODS: ClassVar[frozenset[str]]
+    RETRY_AFTER_STATUS_CODES: ClassVar[frozenset[int]]
+    DEFAULT_REMOVE_HEADERS_ON_REDIRECT: ClassVar[frozenset[str]]
+    DEFAULT_BACKOFF_MAX: ClassVar[int]
+
+    total: bool | int | None
+    connect: int | None
+    read: int | None
+    redirect: Literal[True] | int | None
+    status: int | None
+    other: int | None
+    allowed_methods: Collection[str] | None
+    status_forcelist: Collection[int]
+    backoff_factor: float
+    raise_on_redirect: bool
+    raise_on_status: bool
+    history: tuple[RequestHistory, ...] | tuple[()]
```

Comment:
Why do we need the empty tuple called out here? The first union member is an arbitrary-length tuple, which includes the empty tuple.

Reply:
Oh that's my fault; I suggested that, and didn't think that one through.